### PR TITLE
Split decisions on cell source according to chunks

### PR DESCRIPTION
**Pull request by vidartf**
One-sided merge decisions or two-sided agreement decisions can sometimes contribute to more than one chunk in a string merge. E.g. a patch on line 5 and on line 10 in the cell source would belong to one decision. This would mean that when the user selects e.g. "local" for the patch on line 10, it will also pick "local" for line 5, which is unwanted behavior.

With this change, the web-based mergetool will split these types of decisions such that each decision only contribute to one chunk in the cell source.

### CODE REVIEW

**Comment by vidartf** [comment]
Note that the code in `decisionchunking.ts` is mostly just a translation to typescript of the code in `chunks.py`.

File: nbdime-web/src/chunking/decisionchunking.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+import {
+  hasEntries, valueIn
+} from '../common/util';
+
+
+export
+type MergeChunk = {baseStart: number, baseEnd: number, diffs: DiffCollection}
+
+
+function any(diffs: DiffCollection): boolean {
```
Comment (martinal) [inline nbdime-web/src/chunking/decisionchunking.ts]: IMO "any" is a bit too generic, anyDiffs maybe?

File: nbdime-web/src/chunking/decisionchunking.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    if (e.op === 'addrange') {
+      // Pass
+    } else if (e.op === 'removerange') {
+      k = j + e.length;
+      boundaries.add(k);
+    } else if (e.op === 'patch') {
+      k = j + 1;
+      boundaries.add(k);
+    }
+}
```
Comment (martinal) [inline nbdime-web/src/chunking/decisionchunking.ts]: Confusing indentation of }.

File: nbdime-web/src/chunking/decisionchunking.ts (hunk 1 above)
Comment (vidartf) [inline nbdime-web/src/chunking/decisionchunking.ts]: Agreed.

**Comment by vidartf** [comment]
Comments fixed, and tests passing.

**Pull request merged**